lib/service/Service.ts

```diff
@@ -678,38 +682,64 @@ class Service {
       });
     }
 
-    this.orderBook.on('peerOrder.incoming', order => callback(order));
-    this.orderBook.on('ownOrder.added', order => callback(order));
+    this.orderBook.on('peerOrder.incoming', callback);
+    this.orderBook.on('ownOrder.added', callback);
+
+    const removalCallback = (orderRemoval: OrderPortion) => callback(undefined, orderRemoval);
+    this.orderBook.on('peerOrder.invalidation', removalCallback);
+    this.orderBook.on('peerOrder.filled', removalCallback);
+    this.orderBook.on('ownOrder.filled', removalCallback);
+    this.orderBook.on('ownOrder.removed', removalCallback);
 
-    this.orderBook.on('peerOrder.invalidation', orderRemoval => callback(undefined, orderRemoval));
-    this.orderBook.on('peerOrder.filled', orderRemoval => callback(undefined, orderRemoval));
-    this.orderBook.on('ownOrder.filled', orderRemoval => callback(undefined, orderRemoval));
-    this.orderBook.on('ownOrder.removed', orderRemoval => callback(undefined, orderRemoval));
+    // we wait for the streaming call to be canceled
+    cancelledPromise.then(() => {
```

Comment:
This cleanup won't be required because it will be handled by `takeUntil(cancelled$)`.